Documentation/manual/kcc-design/customization-user-story.md

```diff
@@ -0,0 +1,138 @@
+
+Ideally, our game architecture programmer will be able to use the API of the KCCMovementEngine to make it compatible with our high level code which already handles things such as state switching, etc.
+
+We could probably be convinced to let KCC handle things like gravity for us if the API were well defined and the customization was robust enough, but for the foreseeable future it’s best to think that we are glued to our custom state solution.
+
+**Use Case Level #1: A simple yet robust character that is driven by custom client-side code making use of the Customization Level API**
+
+————————————————————————————————————————————————————————————————————————
+
+**Example Use Case:** For example, a character controller like this is something we would hope to be able to achieve while handling states, input, and events ourselves:
+https://www.youtube.com/watch?v=ylG72jvo_2s&ab_channel=CaptainRipley
+- In fact I intend to start implementing a character controller with equivalent features, that way we could focus on porting it to Open KCC and developing the KCCMovementEngine API alongside it
+
+Features:
+- A Run and Jump State that reads player input and enables planar movement around the scene, as well as custom, client defined jump physics
+- A long Jump state that the player can initiatie with client side input
+    - The author of the long jump state (the customization user)
+
+Technical Requirements:
+- Ability to pass client-side managed velocity values to the KCCMovementEngine for movement in the XYZ plane
+
+- Collision Phase Callbacks, or the ability to read the state of the KCCMovementEngine after executing the move for the frame
+    - Practical Example:
+        - Imagine we have a charge state that starts at a speed of 0 meters per second and then slowly accelerates in a given direction
+        - At one point, our charging character is moving in the positive z axis and has achieved a speed of 15 meters per second
+        - We then call our move function for the frame, but this time we collide head on to a wall
+        - After executing the move function, I should basically be able to read the following data from the KCCMovementEngine Object
+            - “Your desired velocity for the most recent frame was 15, but your actual velocity was lowered to 0 due to the collision phase”
+            - I can then read that 0 value from the KCC Movement Engine, and re-apply it to my Client Side State
+        - This basically keeps the client side velocity and the KCC Movement Engine state in sync, since with this state definition the user should have to Reaccelerate again once more after colliding, they shouldn’t instantly snap back to 15 meters per second if the wall were suddenly removed
+
+- Collision Phase Callbacks Extended
```

Comment:
I like this, right now we can also override the collision data and change how the player moves based on the result of collisions.

I want to change it so that pushing objects works like this as well! I'll add more notes in another PR.